src/App.js

```diff
@@ -1,86 +1,123 @@
-import React from 'react';
+import React, { useEffect, useState, useCallback } from 'react';
+import { NewTodo } from './components/newTodo';
+import { TodoList } from './components/todoList';
+import { FilterTodos } from './components/filterTodos';
+import { useLocalStorage } from './components/localStorage';
 
 function App() {
+  const [todos, setTodos] = useLocalStorage('todosArr');
+  const [uncomplete, setUncomplete] = useState([]);
+  const [toggleAll, setToggleAll] = useState(false);
+  const [filter, setFilter] = useState('All');
+
+  const filterChange = (event) => {
+    setFilter(event.target.name);
+  };
+
+  const changeToggle = (id, completed) => {
+    const index = todos.findIndex(todo => todo.id === id);
+    const updatedTodos = [...todos];
+
+    updatedTodos[index].completed = completed;
+    setTodos(updatedTodos);
+  };
+
+  const addTodo = useCallback((value) => {
+    const todo = {
+      id: +new Date(),
+      title: value,
+      completed: false,
+    };
+
+    if (!value) {
+      return;
+    }
+
+    setTodos([...todos, todo]);
+  }, [todos]);
+
+  const destroyTodo = useCallback((event) => {
+    const { name } = event.target;
+    const index = todos.findIndex(todo => todo.id === +name);
+    const updatedTodos = [...todos];
+
+    updatedTodos.splice(index, 1);
+
+    setTodos(updatedTodos);
+  }, [todos]);
+
+  const clearCompleted = useCallback(() => {
+    const updatedTodos = [...todos].filter(todo => !todo.completed);
+
+    setTodos(updatedTodos);
+    setToggleAll(false);
+  }, [todos]);
+
+  const changeTitle = (id, title) => {
+    const index = todos.findIndex(todo => todo.id === id);
+    const updatedTodos = [...todos];
+
+    updatedTodos[index].title = title;
+    setTodos(updatedTodos);
+  };
```

Comment:
у тебя сложный и не имутабельный подход, ты мутируешь объект (это может вызвать баги), нужно так
```suggestion
  const changeTitle = (id, title) => {
    setTodos( todos.map(todo => { 
       if(todo.id === id) {
         return {
           ...todo,
           title
         };
       }
       
       return todo
    }));
  };
```